Makefile

```diff
@@ -145,7 +145,18 @@ $(FLAGDIR)/%.flags: %/Dockerfile $(FLAG_SH)
 # anything has changed that requires a rebuild.
 #
 $(IMAGE_DIRS): %: %/Dockerfile | check-links
+	@echo
+	@echo "Building [$*] image"
+	@echo
 	cd $(dir $<) && time $(SHELL) -c "( tar -czh . | docker build $(DBFLAGS_$@) -t $@ --label $(LABEL) - )"
+	@echo
+	@echo "Running tests for [$*]"
+	@echo
+	export TESTED_IMAGE=$* && \
+	  export EXPECTED_CAPABILITIES=`cat $*/capabilities.txt` && \
+	  cd test && \
+	  docker-compose up -t 0 -d hadoop-master && \
+	  time docker-compose run --rm test-runner
```

Comment:
[very optional]
I think it should be two separated tasks: all, build and tests, where all could run bulid and tests afterwards